### PR TITLE
Support change variable type code action for collect clause

### PR DESCRIPTION
**Pull request by TharushiJay**
## Purpose
This PR adds the missing change variable type code action for collect clause.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/40554

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by malinthar — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java [hunk 1]
```diff
@@ -151,6 +151,10 @@ private Optional<NonTerminalNode> getVariableOrObjectFieldNode(NonTerminalNode s
             return Optional.of(sNode);
         } else if (isVariableNode(sNode.parent()) || sNode.parent().kind() == SyntaxKind.OBJECT_FIELD) {
             return Optional.of(sNode.parent());
+        } else if (sNode.kind() == SyntaxKind.LIST_CONSTRUCTOR && sNode.parent().kind() == SyntaxKind.COLLECT_CLAUSE
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java:154]: why do we access the parent like this? Is this capture all the possible scenarios?

**Review by LakshanWeerasinghe — CHANGES_REQUESTED** [review]
```ballerina
int letters = from var s in  ["a", "b", "c"] collect ", ".join(s);
```
In here the CA won't work.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java (hunk 1 above)
Comment (TharushiJay) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java]: Updated the logic to be more generic for a collect clause.

**Comment by TharushiJay** [comment]
> ```ballerina
> int letters = from var s in  ["a", "b", "c"] collect ", ".join(s);
> ```
> 
> In here the CA won't work.

Fixed.

**Review by LakshanWeerasinghe — APPROVED** [review]
LGTM!

**Pull request merged**